File: benchmark/nodejs/bfs.js
const n = 100;
const adj = Array.from({ length: n }, () => Array(n).fill(0));
for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
        if ((i * 3 + j * 7) % 11 === 0 && i !== j) adj[i][j] = 1;
    }
}
const start = Date.now();
const visited = Array(n).fill(false);
const queue = [0];
let head = 0;
let count = 0;
while (head < queue.length) {
    const node = queue[head];
    head++;
    if (visited[node]) continue;
    visited[node] = true;
    count++;
    for (let neighbor = 0; neighbor < n; neighbor++) {
        if (adj[node][neighbor] === 1 && !visited[neighbor]) queue.push(neighbor);
    }
}
const end = Date.now();
console.log(`Visited: ${count}`);
console.log(`Time: ${end - start}ms`);

File: benchmark/python/bfs.py
import time
n = 100
adj = [[0] * n for _ in range(n)]
for i in range(n):
    for j in range(n):
        if (i * 3 + j * 7) % 11 == 0 and i != j:
            adj[i][j] = 1
start = time.time() * 1000
visited = [False] * n
queue = [0]
head = 0
count = 0
while head < len(queue):
    node = queue[head]
    head += 1
    if not visited[node]:
        visited[node] = True
        count += 1
        for neighbor in range(n):
            if adj[node][neighbor] == 1 and not visited[neighbor]:
                queue.append(neighbor)
end = time.time() * 1000
print(f"Visited: {count}")
print(f"Time: {end - start:.0f}ms")

